optimize/backend/src/main/java/io/camunda/optimize/service/db/repository/es/TaskRepositoryES.java

```diff
@@ -78,16 +82,42 @@ public boolean tryUpdateByQueryRequest(
       final Query filterQuery,
       final String... indices) {
     LOG.debug("Updating {}", updateItemIdentifier);
+    final boolean clusterTaskCheckingEnabled =
+        configurationService
+            .getElasticSearchConfiguration()
+            .getConnection()
+            .isClusterTaskCheckingEnabled();
+
     final UpdateByQueryRequest updateByQueryRequest =
         UpdateByQueryRequest.of(
             b ->
                 b.index(esClient.addPrefixesToIndices(indices))
                     .query(filterQuery)
                     .conflicts(Conflicts.Proceed)
                     .script(updateScript)
-                    .waitForCompletion(false)
+                    .waitForCompletion(!clusterTaskCheckingEnabled)
                     .refresh(true));
 
+    if (clusterTaskCheckingEnabled) {
+      return asyncUpdate(updateItemIdentifier, filterQuery, updateByQueryRequest);
+    } else {
+      return syncUpdate(updateByQueryRequest);
+    }
+  }
+
+  private boolean syncUpdate(final UpdateByQueryRequest request) {
+    try {
+      final Long deleted = esClient.submitUpdateTask(request).updated();
+      return deleted != null && deleted > 0L;
+    } catch (final IOException e) {
+      throw new OptimizeRuntimeException("Error while trying to submit update task", e);
+    }
+  }
+
+  private boolean asyncUpdate(
```

Comment:
NABD but private methods should come after all public ones

Reply:
fixed